dfastmi/batch/ConfigurationChecker.py

```diff
@@ -27,402 +27,92 @@
-        A discharge and dicharge change determining the discharge exceedance curve (from rivers configuration file) [m3/s].
-    Q : QRuns
-        Tuple of (at most) three characteristic discharges [m3/s].
-    apply_q : Tuple[bool, bool, bool]
-        A list of 3 flags indicating whether each value should be used or not.
-        The Q1 value can't be set to None because it's needed for char_times.
-    tstag : float
-        Fraction of year during which flow velocity is considered negligible [-].
-    T : Vector
-        A vector of values each representing the fraction of the year during which the discharge is given by the corresponding entry in Q [-].
-    rsigma : Vector
-        A vector of values each representing the relaxation factor for the period given by the corresponding entry in Q [-].
-    """
-    q_threshold: Optional[float]
-
-    stages = ApplicationSettingsHelper.get_text("stage_descriptions")
-
-    q_stagnant = reach.qstagnant
-    q_fit = reach.qfit
-    q_levels = reach.qlevels
-    dq = reach.dq
-
-    q_min = reach.qmin
-    try:
-        q_threshold = float(config["General"]["Qthreshold"])
-    except:
-        q_threshold = None
-
-    if q_threshold is None or q_threshold < q_levels[1]:
-        q_bankfull = float(config["General"]["Qbankfull"])
-    else:
-        q_bankfull = 0
-
-    Q, apply_q = dfastmi.kernel.core.char_discharges(q_levels, dq, q_threshold, q_bankfull)
-
-    tstag, T, rsigma = dfastmi.kernel.core.char_times(
-        q_fit, q_stagnant, Q, celerity_hg, celerity_lw, nwidth
-    )
-
-    QList = list(Q)
-    for iq in range(3):
-        if apply_q[iq]:
-            QList[iq] = float(config["Q{}".format(iq + 1)]["Discharge"])
-        else:
-            QList[iq] = None
-    Q = (QList[0], QList[1], QList[2])
-
-    all_q = True
-    return (
-        all_q,
-        q_threshold,
-        q_bankfull,
-        q_fit,
-        Q,
-        apply_q,
-        tstag,
-        T,
-        rsigma,
-    )
 
-def check_configuration_v1_cond(config: configparser.ConfigParser, mode_str: str, i : int) -> bool:
-    """
-    Check validity of one condition of a version 1 analysis configuration.
-
-    Arguments
-    ---------
-    config : configparser.ConfigParser
-        Configuration for the D-FAST Morphological Impact analysis.
-    mode_str : str
-        String indicating the source of the model data: WAQUA or D-Flow FM.
-    i : int
-        Flow condition to be checked.
-
-    Returns
-    -------
-    success : bool
-        Boolean indicating whether the D-FAST MI analysis configuration is valid.
-    """
-    cond = "Q" + str(i+1)
-    if mode_str == WAQUA_EXPORT:
-        # condition block may not be specified since it doesn't contain any required keys
-        if cond in config.sections() and "Discharge" in config[cond]:
-            # if discharge is specified, it must be a number
-            if not is_float_str(config[cond]["Discharge"]):
+        hydro_q = reach.hydro_q
```

Comment:
should there be a check that reach is a ReachAdvanced?

Reply:
maybe _get_reach can also be overridden and a check be added if the reach is the expected type.
E.g. something like:

````
def _get_reach(self, rivers : RiversObject, config : configparser.ConfigParser) -> AdvancedReach:
      reach = super()._protected_method()
      if not isinstance(reach , AdvancedReach)
          raise typeerror(insert message here)
      return reach
````

Reply:
idk how valid the above piece of code is and if pydentic could handle the instance check instead.
Lets discuss this further, might also be relevant for the legacy version.

Reply:
I have made the base method generic. the derived call can call with the reach type they expect to be returned